code/game/objects/items/devices/uplinks.dm

```diff
@@ -186,6 +186,10 @@ A list of items and costs is stored under the datum of every game mode, alongsid
 		if (uses <= 0)
 			return
 		var/amount = input("How many telecrystals do you wish to withdraw?:", "Extract telecrystals", null) as num
+		if(amount <= 0)
```

Comment:
should've used clamp and adjacent reee

Reply:
it shouldvent been taken out of its draft....time for a revert.....